packages/geolocator/tizen/src/geolocator_tizen_plugin.cc

```diff
@@ -14,25 +14,119 @@
 #include <memory>
 #include <string>
 
-#include "locaton_manager.h"
+#include "app_settings_manager.h"
+#include "location_manager.h"
 #include "log.h"
 #include "permission_manager.h"
-#include "setting.h"
 
 namespace {
 
+typedef flutter::MethodChannel<flutter::EncodableValue> FlMethodChannel;
+typedef flutter::EventChannel<flutter::EncodableValue> FlEventChannel;
+typedef flutter::MethodCall<flutter::EncodableValue> FlMethodCall;
+typedef flutter::EventSink<flutter::EncodableValue> FlEventSink;
+typedef flutter::MethodResult<flutter::EncodableValue> FlMethodResult;
+typedef flutter::StreamHandler<flutter::EncodableValue> FlStreamHandler;
+typedef flutter::StreamHandlerError<flutter::EncodableValue>
+    FlStreamHandlerError;
+
+constexpr char kPrivilegeLocation[] = "http://tizen.org/privilege/location";
+
+class LocationStreamHandler : public FlStreamHandler {
+ protected:
+  std::unique_ptr<FlStreamHandlerError> OnListenInternal(
+      const flutter::EncodableValue *arguments,
+      std::unique_ptr<FlEventSink> &&events) override {
+    events_ = std::move(events);
+    LocationCallback callback = [this](Position position) -> void {
+      events_->Success(position.ToEncodableValue());
+    };
+    try {
+      location_manager_.StartListenLocationUpdate(callback);
+    } catch (const LocationManagerError &error) {
+      return std::make_unique<FlStreamHandlerError>(
+          std::to_string(error.GetErrorCode()), error.GetErrorString(),
+          nullptr);
```

Comment:
This part is causing a hard crash (SIGABRT) due to the issue mentioned in https://github.com/flutter-tizen/plugins/pull/351#discussion_r847005275. battery_plus_tizen and connectivity_plus_tizen also had the same problem but we ignored because the APIs used by the plugins had very low chance of getting an error, but unfortunately it's not the case for this plugin (e.g. you will get an error if a permission is not granted to the app).

You may modify the code as follows for now (until the issue gets resolved):
```suggestion
      LOG_ERROR("Failed to set a location update callback: %s",
                error.GetErrorString().c_str());
      return nullptr;
```
I'll check if there's any better way to handle this.

Reply:
How about this?

```cpp
    } catch (const LocationManagerError &error) {
      // This is an intentional leak.
      // Issue: https://github.com/flutter/flutter/issues/101682
      std::string *code = new std::string(std::to_string(error.GetErrorCode()));
      std::string *message = new std::string(error.GetErrorString());
      return std::make_unique<FlStreamHandlerError>(*code, *message, nullptr);
    }
```